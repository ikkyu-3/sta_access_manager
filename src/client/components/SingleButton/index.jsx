// @flow
import type { SingleButtonProps } from "@types";

import * as React from "react";
import styles from "./style.css";

function SingleButton(props: SingleButtonProps): React.Node {
  const { text = "", className = "default", onButtonClick } = props;
  const { singleButton } = styles;

  return (
    <div className={singleButton}>
      <button className={styles[className]} onClick={onButtonClick}>
        {text}
      </button>
    </div>
  );
}
export default SingleButton;
