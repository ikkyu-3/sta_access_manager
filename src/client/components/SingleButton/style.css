@charset "utf-8";
.singleButton {
  position: absolute;
  width: 100%;
  height: 70px;
  bottom: 0;
}

.singleButton > button {
  width: 100%;
  height: 100%;
  margin: 0;
  padding: 0;
  border: none;
  font-size: 24px;
  font-weight: bold;
}

.singleButton > button:focus {
  outline: none;
}

.default {
  background-color: #00bcd4;
  color: whitesmoke;
}

.next {
  background-color: orange;
  color: whitesmoke;
}
