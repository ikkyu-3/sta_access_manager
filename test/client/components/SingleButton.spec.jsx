import React from "react";
import renderer from "react-test-renderer";
import { shallow } from "enzyme";
import SingleButton from "@components/SingleButton";

const fn = () => {};

describe("SingleButton.jsxのテスト", () => {
  describe("スナップショット", () => {
    it("正しいレンダリング", () => {
      const tree = renderer.create(
        <SingleButton
          text="Single Button"
          className="default"
          onButtonClick={fn}
        />
      );
      expect(tree).toMatchSnapshot();
    });
  });

  describe("コンポーネントのテスト", () => {
    const onClick = jest.fn();
    const button = shallow(
      <SingleButton text="Single Button" onButtonClick={onClick} />
    ).find("button");

    it("classNameを指定しない時、ボタン表示はdefaultスタイルである", () => {
      expect(button.hasClass("default")).toBeTruthy();
    });

    it("textに渡した値が、文表示される", () => {
      expect(button.text("Single Button")).toBeTruthy();
    });

    it("ボタンをクリックすると、onButtonClickに渡した関数が実行される", () => {
      button.simulate("click");
      expect(onClick).toBeCalled();
    });

    it("classNameにnextと指定した時、ボタン表示はnextスタイルである", () => {
      const nextButton = shallow(
        <SingleButton text="Next" className="next" onButtonClick={fn} />
      );
      expect(nextButton.find("button").hasClass("next")).toBeTruthy();
    });
  });
});
