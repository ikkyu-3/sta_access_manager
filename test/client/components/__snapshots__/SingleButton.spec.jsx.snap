// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SingleButton.jsxのテスト スナップショット 正しいレンダリング 1`] = `
<div
  className="singleButton"
>
  <button
    className="default"
    onClick={[Function]}
  >
    Single Button
  </button>
</div>
`;
